### ISSUE TITLE
[Bug?]: Setting multiple Set-Cookie headers is not handled properly on undici version >=5.19.0

### ISSUE THREAD

**Issue by GiyoMoon** [solidjs/solid-start#953]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Since `undici` version 5.19.0, setting multiple cookies in `createServerAction$` with the `Set-Cookie` header isn't possible anymore. It only returns the last(?) cookie which was appended to the header.

I hope I have submitted this issue to the correct repository. If I should open this issue on the `undici` project instead, feel free to close the issue.

### Expected behavior 🤔

The server action should return all set cookies in the response like described in #656

### Steps to reproduce 🕹

I created an example here: https://github.com/GiyoMoon/solid-start-multi-cookie

Steps to reproduce:
```
git clone https://github.com/GiyoMoon/solid-start-multi-cookie
cd solid-start-multi-cookie
pnpm install
pnpm dev
```
Navigate to http://localhost:3000 and call the server action with the `Set cookies` button. If you check the response of the http call you just made, it only contains one `Set-Cookie` header.
If you check the cookie store, it only contains `cookie-2`, but it should also contain `cookie-1`.

If you downgrade to `undici` version `5.18.0`, it works again.

### Context 🔦

I'm trying to set two cookies with different `maxAge` values, and therefore I have to use two separete stores which means two separate cookies.

### Your environment 🌎

```shell
System: Apple M1 Pro v13.4.1
Nodejs: v18.16.1
```

**Comment by edivados** [comment]
In undici 5.19.0
Set-Cookie appears multiple times in the header. Server iterates over all `set-cookie` keys and only the last one will be set. Splitting does nothing here as the cookies are already splitted.

In undici 5.18.0
Set-Cookie appears only once in the header. Server splits the string and setHeader works as expected when passing in an array.

Changing `res.setHeader` to `res.appendHeader` should fix it.

https://github.com/solidjs/solid-start/blob/52d44e2cad33faf3dc40148520d7ae41086dd6a8/packages/start/node/fetch.js#L160-L164

**Comment by birkskyum** [comment]
With the latest solid-start 0.3.1 this gives me an:

```sh
An unhandled error occured: TypeError: res.appendHeader is not a function
    at handleNodeResponse (file:///Users/admin/repos/regenworks-isomorphic/node_modules/solid-start/node/fetch.js:168:11)
    at startHandler (file:///Users/admin/repos/regenworks-isomorphic/node_modules/solid-start/dev/server.js:138:7)
```

@ryansolid ,  did something related to undici / node fetch change with the 0.3.1 release?

**Comment by edivados** [comment]
@birkskyum What node version are you using? I tested on node lts (18.12.0) and above and I can imagine it not existing on lower versions.

**Comment by birkskyum** [comment]
@edivados I experience it on node 18 and 20. On node 16 i get on startup:

```sh
 solid-start dev 
 version  0.3.1
 adapter  node
file:///Users/admin/repos/regenworks-isomorphic/node_modules/solid-start/node/fetch.js:73
export class NodeRequest extends Request {
                                 ^

ReferenceError: Request is not defined
    at file:///Users/admin/repos/regenworks-isomorphic/node_modules/solid-start/node/fetch.js:73:34
```

**Comment by birkskyum** [comment]
@edivados , here is a repro: https://github.com/birkskyum/solid-start-https-appendHeader-0.3.1

I found that my vite.config.ts are causing this:

```ts
import solid from "solid-start/vite";
import { defineConfig } from "vite";
import mkcert from "vite-plugin-mkcert";

export default defineConfig({
  server: { https: true, host: "127.0.0.1", port: 3000 },
  plugins: [
    mkcert(),
    solid({
      ssr: true,
    }),
  ]
});
```

If I set the `https: false` suddenly the appendHeader can be found, so I wonder why the https is suddenly breaking.

**Comment by edivados** [comment]
@birkskyum Hmmm https is maybe a separate issue. According to [nodejs](https://nodejs.org/api/http.html#outgoingmessageappendheadername-value) appendHeader was adden in v18.3.0, v16.17.0 and in fact 18.2.0 does not work. I can't reproduce the issue with 20.0.0. I am using OPs repo with solid-start 0.2.31 for testing.

Yeah... 16.x has the same Request error for me. I wonder how the tests pass on GitHub?

Regarding appendHeader it is probably better to revert back to setHeader and solve this another way. Maybe something like this?
Also adding tests... 😭 

```js
const cookies = [];

for (const [name, value] of webRes.headers) {
    if (name === "set-cookie") {
      cookies.push(...splitCookiesString(value));
    } else res.setHeader(name, value);
}

if (cookie.length) res.setHeader(cookies);

```

**Comment by birkskyum** [comment]
@edivados I only experience this in 0.3.1 - haven't tried running the OP repo that uses solid-start version 0.2.29. I use 0.2.29  in production, where https works fine.

**Comment by ryansolid** [comment]
I dropped support for 16 in 0.3.x and switched to Node. Is `appendHeader` not there in 18 then?

**Comment by edivados** [comment]
@ryansolid I think the issue is that when setting https certs in vite Http2ServerResponse gets used and maybe there is no appendHeader. I am still looking at it so not 100% sure.

**Comment by birkskyum** [comment]
@edivados , I think you're on to something there. It does seem like it's only present in the [http](https://nodejs.org/api/http.html) response and not in [http2](https://nodejs.org/api/http2.html#responsegetheadername) response

<img width="349" alt="Screenshot 2023-08-09 at 21 34 50" src="https://github.com/solidjs/solid-start/assets/74932975/def9a6e5-bafa-43d8-a914-720fb69677ea">
<img width="427" alt="Screenshot 2023-08-09 at 21 34 57" src="https://github.com/solidjs/solid-start/assets/74932975/14ec61a4-7069-4248-ba5d-7f7ae1563b07">

**Comment by edivados** [comment]
@birkskyum Ok.. so #955 breaks using https in vite because there is no appendHeader in Http2ServerResponse. 😢 
I tested your repo with 0.2.29 and 0.2.30. 0.2.30 does not work as expected.
